pytorch_lightning/plugins/training_type/parallel.py

```diff
@@ -98,7 +100,17 @@ def block_backward_sync(self):
         This is useful for skipping sync when accumulating gradients, reducing communication overhead
         Returns: context manager with sync behaviour off
         """
-        if isinstance(self.model, LightningDistributedDataParallel):
+        if isinstance(self.model, (LightningDistributedDataParallel, DistributedDataParallel)):
             yield self.model.no_sync()
         else:
             yield None
+
+    def broadcast(self, obj: object, src: int) -> object:
```

Comment:
We might want to update LightningModule `all_gather` to rely on this broadcast, we can transfer none tensor objects.